developerportal/templates/molecules/search-result.html

```diff
@@ -0,0 +1,39 @@
+{% load wagtailcore_tags %}
+{% load app_filters %}
+
+<li class="mzp-c-card mzp-c-card-medium search-result-item">
+  <a
+    href="{% pageurl page %}"
+    class="mzp-c-card-block-link
+    {% if page.resource_type == "video" and page.is_external %}
+      js-modal-trigger{% endif %}"
+    {% if page.is_external %}
+      target="_blank"
+      rel="nofollow noopener"
+      data-title="{{ page.title }}"
+      data-type="{{ page.resource_type }}"
+    {% endif %}
+  >
+  <h4>
+      <span class="search-result-title">{{ page.title }}</span>
+      {% with page|page_type_label as label %}
+        {% if label %}
+          <span class="search-result-type">{{label}}</span>
+        {% endif %}
+      {% endwith %}
+    </h4>
```

Comment:
Do we need both `span` elements? Could the main styling be on the `h4` and then the nested `span` overrides as needed?

Reply:
True! Will revisit!

Reply:
Ah ha - I remembered/rediscovered why i needed the two spans -- it was to stop the reflow breaking down on mobile viewport:

Without the span for the title text, I have to put `left-margin` on the `span.label` instead to space it away from the H4 main text. For example, in med and large viewports:

<img width="586" alt="Screenshot 2020-06-30 at 22 40 00" src="https://user-images.githubusercontent.com/101457/86179688-acc38b80-bb22-11ea-860c-b621c9eb9181.png">

but this breaks down particularly on the smallest viewport if the margin is on the `span.label` rather than on a span wrapping the main h4's text:

No  `span.search-result-title`:

<img width="386" alt="Screenshot 2020-06-30 at 22 35 05" src="https://user-images.githubusercontent.com/101457/86179807-dd0b2a00-bb22-11ea-9db8-ba430b3ad3c0.png">


With `span.search-result-title`:
<img width="319" alt="Screenshot 2020-06-30 at 22 35 31" src="https://user-images.githubusercontent.com/101457/86179825-e4323800-bb22-11ea-8305-5d4e98d360d9.png">

Reply:
Could we have:

```
span.search-result-title {
    margin-left: 0;

    mq-tablet-and-up {
	    margin-left: 10px;
    }
}
```

Reply:
@schalkneethling Tried it but still not quite idea, because smallest viewport ends up with not enough left-margin for the labels
<img width="252" alt="Screenshot 2020-07-01 at 23 14 11" src="https://user-images.githubusercontent.com/101457/86296506-d5b25200-bbf0-11ea-915a-23f74ed47224.png">
 

Am thinking I might have to just live with this markup for now, if that's OK. Can ticket it to improve when I have more time